### PR TITLE
Complementary filter for velocities

### PR DESCRIPTION
**Pull request by amiller27**

### CODE REVIEW

**Comment by amiller27** [comment]
Not quite ready, give me a minute

**Comment by amiller27** [comment]
Tested and ready to merge

File: src/iarc7_sensors/vel_filter.py [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+            assert last_differential_i < i and last_differential_i >= -1
+
+            if klass == 'kalman':
+                if last_differential_i == -1:
+                    # No other differential measurements in the queue, so
+                    # assume our velocity hasn't changed since the previous
+                    dx = 0.0
+                    dy = 0.0
+                else:
+                    last_diff_vector = \
```
Comment (asaba96) [inline src/iarc7_sensors/vel_filter.py:185]: What is this operation doing? I've never seen that before. @amiller27

Comment (amiller27) [reply src/iarc7_sensors/vel_filter.py:185]: Backslash is line continuation in Python, it just means the next line isn't a new statement. You only need them if you aren't currently inside parens or brackets.

Comment (asaba96) [reply src/iarc7_sensors/vel_filter.py:185]: Oh okay makes sense.

**Review by asaba96 — APPROVED** [review]
Looks good to me.

**Pull request merged**